web/packages/teleterm/src/ui/DocumentCluster/UnifiedResources.tsx

```diff
@@ -0,0 +1,287 @@
+        props.clusterUri,
+      ]
+    ),
+  });
+
+  useEffect(() => {
+    const { cleanup } = onResourcesRefreshRequest(() =>
+      fetch({
+        force: true,
+        fromStart: true,
+      })
+    );
+    return cleanup;
+  }, [onResourcesRefreshRequest, fetch]);
+
+  return (
+    <SharedUnifiedResources
+      params={params}
+      setParams={setParams}
+      updateUnifiedResourcesPreferences={() => alert('Not implemented')}
+      onLabelClick={() => alert('Not implemented')}
+      pinning={{ kind: 'hidden' }}
+      resources={resources
+        .map(resource => {
+          switch (resource.kind) {
+            case 'server': {
+              const { resource: server } = resource;
+              const serverId = routing.parseServerUri(server.uri)?.params
+                ?.serverId;
+              if (!serverId) {
+                return;
+              }
+              return {
+                resource: {
+                  kind: 'node' as const,
+                  labels: server.labelsList,
+                  id: serverId,
+                  name: server.name,
+                  hostname: server.hostname,
+                  addr: server.addr,
+                  tunnel: server.tunnel,
+                },
+                ui: {
+                  ActionButton: <ConnectServerActionButton server={server} />,
+                },
+              };
+            }
+            case 'database': {
+              const { resource: database } = resource;
+              return {
+                resource: {
+                  kind: 'db' as const,
+                  labels: database.labelsList,
+                  description: database.desc,
+                  name: database.name,
+                  type: database.type,
+                  protocol: database.protocol as DbProtocol,
+                },
+                ui: {
+                  ActionButton: (
+                    <ConnectDatabaseActionButton database={database} />
+                  ),
+                },
+              };
+            }
+            case 'kube': {
+              const { resource: kube } = resource;
+
+              return {
+                resource: {
+                  kind: 'kube_cluster' as const,
+                  labels: kube.labelsList,
+                  name: kube.name,
+                },
+                ui: {
+                  ActionButton: <ConnectKubeActionButton kube={kube} />,
+                },
+              };
+            }
+          }
+        })
+        .filter(Boolean)}
+      resourcesFetchAttempt={attempt}
+      fetchResources={fetch}
+      availableKinds={['db', 'kube_cluster', 'node']}
+      Header={pinAllButton => (
+        <Flex alignItems="center" justifyContent="space-between">
+          {/*temporary search panel*/}
+          <SearchPanel
+            params={params}
+            pathname={''}
+            replaceHistory={() => undefined}
+            setParams={setParams}
+          />
```

Comment:
I ended up also "no-oping" the `pathname` and `replaceHistory` in access requests. Perhaps this component can be remade to make them optional. Although yours is temporary so maybe it's only a problem for access requests (which in itself is _also_ temporary until it gets moved to the main view)

Reply:
Instead of making them optional, it would be even better to remove them completely. 
The `useUrlFiltering` hook should take care of replacing the history with the correct pathname. Children shouldn't know about `pathname` or `replaceHistory`.

I was thinking about refactoring it while I was doing `UnifiedResources` refactor, but eventually I left it as is (I wanted to limit the scope of changes).